sdk/dotnet/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -121,7 +125,7 @@ void LogExcessive(string message)
         }
 
         private static Task<ImmutableArray<Resource>> GatherExplicitDependenciesAsync(InputList<Resource> resources)
-            => resources.ToOutput().GetValueAsync();
+            => resources.ToOutput().GetValueAsync(whenUnknown: ImmutableArray<Resource>.Empty);
```

Comment:
This change in behavior is intentional.

I have been getting an issue on a very large stack privately shared by @gitfool that is difficult to minimize, but essentially it was hitting this line in preview and failing with:

```
System.InvalidOperationException: This operation cannot be performed on a default instance of ImmutableArray<T>.  Consider
initializing the array, or checking the ImmutableArray<T>.IsDefault property.
  at System.Collections.Immutable.ImmutableArray`1.ThrowInvalidOperationIfNotInitialized()
  at System.Collections.Immutable.ImmutableArray`1.System.Collections.Generic.ICollection<T>.get_Count()
  at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
  at Pulumi.Deployment.PrepareResourceAsync(String label, Resource res, Boolean custom, Boolean remote, ResourceArgs args,
     ResourceOptions options) in Deployment_Prepare.cs:24
  at Pulumi.Deployment.RegisterResourceAsync(Resource resource, Boolean remote, Func`2 newDependency, ResourceArgs args,
     ResourceOptions options) in Deployment_RegisterResource.cs:27
  at Pulumi.Deployment.ReadOrRegisterResourceAsync(Resource resource, Boolean remote, Func`2 newDependency, ResourceArgs args
     , ResourceOptions options) in Deployment_ReadOrRegisterResource.cs:73
  at Pulumi.Deployment.CompleteResourceAsync(Resource resource, Boolean remote, Func`2 newDependency, ResourceArgs args,
     ResourceOptions options, ImmutableDictionary`2 completionSources) in Deployment_ReadOrRegisterResource.cs:89
  at Pulumi.Output`1.GetValueAsync() in Output.cs:153
  at Pulumi.Deployment.EngineLogger.TryGetResourceUrnAsync(Resource resource) in Deployment.EnginerLogger.cs:144
  at Pulumi.Deployment.<>c__DisplayClass63_0.<<RunInlineAsync>b__1>d.MoveNext() in Deployment_Inline.cs:62

```